demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c

```diff
@@ -0,0 +1,238 @@
+     * For legacy reasons, Cryptoki demands that the CKF_SERIAL_SESSION bit
+     * is always set. */
+    xResult = pxFunctionList->C_OpenSession( pxSlotId[ 0 ],
+                                             CKF_SERIAL_SESSION | CKF_RW_SESSION,
+                                             NULL, /* Application defined pointer. */
+                                             NULL, /* Callback function. */
+                                             &hSession );
+    assert( xResult == CKR_OK );
+
+
+    /* C_Login is called to log the user in to the token. The login status is
+     * shared between sessions, so logging in once is sufficient for all the sessions
+     * tied to the token. Most of the behavior for C_Login is defined by the token
+     * so it may be necessary to modify calls to C_Login when switching to a different
+     * Cryptoki library or token.
+     *
+     * This Cryptoki library does not implement C_Login, and only defines the function
+     * for compatibility reasons.
+     */
+    xResult = pxFunctionList->C_Login( hSession,
+                                       CKU_USER,
+                                       ( CK_UTF8CHAR_PTR ) "0000",
+                                       sizeof( "0000" ) - 1UL );
+    assert( xResult == CKR_OK );
+
+    /* C_GenerateRandom generates random or pseudo random data. As arguments it
+     * takes the application session, and a pointer to a byte buffer, as well as
+     * the length of the byte buffer. Then it will fill this buffer with random
+     * bytes. */
+    xResult = pxFunctionList->C_GenerateRandom( hSession,
+                                                xRandomData,
+                                                sizeof( xRandomData ) );
+    assert( xResult == CKR_OK );
+
+    for( ulIndex = 0; ulIndex < sizeof( xRandomData ); ulIndex++ )
+    {
+        LogInfo( ( "Generated random number: %x", xRandomData[ ulIndex ] ) );
+    }
+
+    /* C_CloseSession closes the session that was established between the
+     * application and the token. This will clean up the resources that maintained
+     * the link between the application and the token. If the application wishes
+     * to use the token again, it will need to open a new session. */
+    xResult = pxFunctionList->C_CloseSession( hSession );
+    assert( xResult == CKR_OK );
+
+    /* C_Finalize signals to the Cryptoki library that the application is done
+     * using it. It should always be the last call to the Cryptoki library.
+     * NULL should always be passed as the argument, as the parameter is currently
+     * just reserved for future revisions.
+     *
+     * Calling this function in a multi threaded environment can lead to undefined
+     * behavior if other threads are accessing the Cryptoki library. */
+    xResult = pxFunctionList->C_Finalize( NULL );
+    assert( xResult == CKR_OK );
+
+    LogInfo( ( "Finished PKCS #11 Management and Random Number Generation" \
+               " Demo." ) );
```

Comment:
Escapes are not required for macro calls!
```suggestion
    LogInfo( ( "Finished PKCS #11 Management and Random Number Generation" 
               " Demo." ) );
```